### PR TITLE
[New Rule] Adding Lateral Movement Rules from Advanced Analytic LMD Package

### PR DESCRIPTION
**Pull request by ajosh0504**
## Issues
Related https://github.com/elastic/ia-trade-team/issues/174

## Summary
This PR moves the detection rules from the Lateral Movement Detection [integration](https://github.com/elastic/integrations/tree/main/packages/lmd/kibana/security_rule) package, into the detection-rules repo.

### CODE REVIEW

**Comment by ajosh0504** [comment]
Note that some of the rules in this PR reference non-ECS fields, the mappings for which can be found [here](https://github.com/elastic/integrations/blob/main/packages/lmd/elasticsearch/transform/pivot_transform/fields/fields.yml).

**Comment by terrancedejesus** [comment]
per [481a9e92301d2ecdc1c01d829d695bb92f2092bf](https://github.com/elastic/detection-rules/pull/3119/commits/481a9e92301d2ecdc1c01d829d695bb92f2092bf) I pushed, the following has been adjusted:

- added `endpoint` integration tag to rules with EQL queries and respective indices (2 rules)
- built the integration manifests and integration schemas, adding `lmd` to the existing
- added `--integration` argument to `build-schemas` CLI command to speed up time for adding and updating single integration schemas
- removed `required=True` from `IntegrationManifestSchema` class as `lmd` does not have any policy templates so this is not required upstream in packages. `transform_policy_template` will still execute as a post load method, but check for policy templates not and if not ignore. Remember that policy templates are technically the integration (cloudtrail) and the package (aws) are used for related integrations.
- adjusted `build_intefgrations_schemas()` method in `devtools.py` so it  allowed either a single integration schema to be built and added or all of them by default as usual.
- added `ANALYTIC_PACKAGES` to `definitions.py` as these are exceptions to some of the unit tests regarding integrations, tags and indices.
- adjusted unit tests to account for analytic package exceptions that do not apply to the existing constraints and requirements.

## Testing
Below are some screenshots regarding the testing.

**Ensure that related integrations tag is correct when rule (API format) is built**
Screenshot 2023-09-22 at 3 37 03 PM

**Ensure that the `lmd` schema is loading and being used for rule validation**
Screenshot 2023-09-22 at 2 18 56 PM

**Built a package, filtered NDJSON rules and successfully loaded into stack**
Screenshot 2023-09-22 at 3 04 14 PM

**Rules that are loaded, reference the `lmd` package as expected**
Screenshot 2023-09-22 at 2 05 14 PM
Screenshot 2023-09-22 at 2 04 13 PM

**Changes made do not alter existing rule versions for other rules accidentally**
Screenshot 2023-09-22 at 1 59 46 PM

File: detection_rules/devtools.py
Comment (Mikaayenson) [inline detection_rules/devtools.py:1240]: YES!

File: detection_rules/integrations.py
Comment (Mikaayenson) [inline detection_rules/integrations.py:50]: Does this mean not all integrations have a policy_template? Should they?

File: detection_rules/schemas/definitions.py
Comment (Mikaayenson) [inline detection_rules/schemas/definitions.py:36]: Will this list grow to basically be NON_DATASET_PACKAGES + ANALYTIC_PACKAGES?

File: tests/test_all_rules.py
Comment (Mikaayenson) [inline tests/test_all_rules.py:636]: So none of these will be tagged?

File: rules/integrations/lmd/lateral_movement_malicious_remote_file_creation.toml
Comment (Mikaayenson) [inline rules/integrations/lmd/lateral_movement_malicious_remote_file_creation.toml:20]: Are these new UUIDs or duplicates from the original rules?

File: detection_rules/integrations.py
Comment (ajosh0504) [reply detection_rules/integrations.py:50]: Sorry for the noob q, but what does a policy template mean?

Comment (ajosh0504) [reply detection_rules/integrations.py:50]: Confirming that not all integrations have policy templates, so this is a valid change.

Comment (terrancedejesus) [reply detection_rules/integrations.py:50]: @Mikaayenson policy templates are not a requirement for packages, so when we originally made this dataclass we may have added more constraints than required. @ajosh0504 policy templates in the manifest are how Fleet packages distinguish between a package (AWS) and it's actual integrations (CloudTrail). So for example, in the AWS package, CloudTrail would have its own section (policy template) in the AWS manifest. 

Endpoint is a similar package/integration, but the package `endpoint` has the `endpoint` policy template in it. The packages are still valid without a policy template so no need to adjust the analytic packages manifest.

File: detection_rules/schemas/definitions.py
Comment (terrancedejesus) [reply detection_rules/schemas/definitions.py:36]: @Mikaayenson yes I was thinking about this.

The `NON_DATASET_PACKAGES` are packages that are the following:

- valid packages with a manifest
- have policy templates that match the package name
- no separate integrations

The `ANALYTIC_PACKAGES` are specifically packages by the ML team only.

- valid packages with a manifest
- have no policy template
- no separate integrations

Code logic wise, `ANALYTIC_PACKAGES` is referenced in unit testing as these are exceptions for the rule tags and index requirements defined in the rule. We now have rules that may have `lmd` as a tag, but no direct correlation based on index defined and thus will fail unit testing.

`NON_DATASET_PACKAGES`, are used as an exception when adding `related_integrations` data to the rule during API conversion.

File: tests/test_all_rules.py
Comment (terrancedejesus) [reply tests/test_all_rules.py:636]: @Mikaayenson the rules will be tagged, for example with `lmd` but are not required to have a matching index pattern defined in the rule. This is because the data is typically reliant on, for say, `endpoint` (`log-endpoint.events*`. But we need the `lmd` tag to build the `related_integrations` field and to do field validation with the schemas.

File: detection_rules/schemas/definitions.py
Comment (terrancedejesus) [reply detection_rules/schemas/definitions.py:36]: combine lists for analytic package and non dataset when doing logic for related integrations field.

**Comment by terrancedejesus** [comment]
@Mikaayenson @eric-forte-elastic - Adjusted the static definitions in `definitions.py` so that in `rule.py` ineligible rules are simply determined by combining both lists.

Ready for review!

**Review by Mikaayenson — APPROVED** [review]
LGTM!

**Review by eric-forte-elastic — APPROVED** [review]
LGTM too! Great walk-through earlier :+1:

**Comment by ajosh0504** [comment]
@terrancedejesus Looking great! 🥳 Could you also confirm that nothing more needs to happen in the integrations [PR](https://github.com/elastic/integrations/pull/7901) to remove these rules?

**Comment by terrancedejesus** [comment]
@ajosh0504 nothing that I can see in the integrations package. I'd hold off on merging until Monday. We can then sync our package update with this package update to minimize the feature unavailability because rules would be missing. If

**Comment by ajosh0504** [comment]
Haha no, agreed. Let's wait until Monday and merge simultaneously!

**Comment by terrancedejesus** [comment]
@ajosh0504 - merging this in once checks are complete. We will not be releasing a package until 10/02/2023 so when we release we can merge your PR into integrations.

Release reference: https://github.com/elastic/ia-trade-team/issues/185

**Pull request merged**
